documentation/src/docs/basic-reader-and-writer.md

```diff
@@ -9,243 +9,196 @@ You may obtain a copy of the License at
-the context of a Scope, since all Readers, Writers and other Clients created by
-the ClientFactory are created in the context of that Scope.  The ClientFactory
-also needs a URI to one of the Pravega Controllers, just like StreamManager.
+A `ClientFactory` is created in the context of a Scope, since all Readers, Writers and other Clients created by the `ClientFactory` are created in the context of that Scope. The `ClientFactory`
+also needs a URI to one of the Pravega Controllers (`ClientFactory.withScope(scope, controllerURI)`) , just like `StreamManager`.
+
+As the `ClientFactory` and the objects it creates consume resources from Pravega and implement `AutoCloseable`, it is a good practice to create these objects using a try-with-resources. By doing this, we make sure that, regardless of how the application ends, the Pravega resources will be properly closed in the right order.
+
 
-Because ClientFactory and the objects it creates consumes resources from
-Pravega, it is a good practice to create these objects in a try-with-resources
-statement.  Because ClientFactory and the objects it creates all implement
-Autocloseable, the try-with-resources approach makes sure that regardless of how
-your application ends, the Pravega resources will be properly closed in the
-right order.
+Once the `ClientFactory` is instantiated, we can use it to create a Writer. There are
+several things a developer needs to know before creating a Writer:
 
-Now that we have a ClientFactory, we can use it to create a Writer.  There are
-several things a developer needs to know before s/he creates a Writer:
+1.  _What is the name of the Stream to write to?_
+    (The Scope has already
+    been determined when the `ClientFactory` was created.)
 
-1.  What is the name of the Stream to write to?  Note: the Scope has already
-    been determined when the ClientFactory was created
+2.  _What Type of Event objects will be written to the Stream?_
 
-2.  What Type of Event objects will be written to the Stream?
+3.  _What serializer will be used to convert an Event object to bytes?_ (Recall
+    that Pravega only knows about sequences of bytes, it is unaware about Java objects.)
 
-3.  What serializer will be used to convert an Event object to bytes?  Recall
-    that Pravega only knows about sequences of bytes, it does not really know
-    anything about Java objects.
+4.  _Does the Writer need to be configured with any special behavior?_
 
-4.  Does the Writer need to be configured with any special behavior?
 
-In our example, lines 11-13 show all these decisions.  This Writer writes to the
-Stream specified in the configuration of the HelloWorldWriter object itself (by
-default the stream is named "helloStream" in the "examples" Scope).  The Writer
+```Java
+     EventStreamWriter<String> writer = clientFactory.createEventWriter(streamName,
+                                                      new JavaSerializer<String>(),
```

Comment:
This isn't a great pattern. We have a `UTF8StringSerializer` which is specific to strings and is better for them. Java serialize is a basic fallback for any other random Java object that you don't want to write a serializer for. In general many users will want to write serializes or use our Json and Avro serializes which are located here: https://github.com/pravega/pravega-serializer

Reply:
@sandeep @RaulGracia  please record your comments on `javaSerializer`

Reply:
The following is added in the content
` **Note:** Pravega allows users to write their own serializer. For more information and example, please refer to [Pravega Serializer](https://github.com/pravega/pravega-serializer). `

Reply:
@tkaitchuck @shrids 
Issue created : https://github.com/pravega/pravega-serializer/issues/5